From: Dmitry Shachnev <mitya57@debian.org>
Date: Mon, 29 Jun 2020 15:08:42 +0300
Subject: Preload libGLX_mesa.so.0 in web_engine_initialize.py

We need to load this library before loading QtWebEngine, because
otherwise loading it will fail on arm64 with the following error:

/usr/lib/aarch64-linux-gnu/libglapi.so.0: cannot allocate memory in static TLS block
---
 sources/pyside2/tests/QtWebEngine/web_engine_initialize.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sources/pyside2/tests/QtWebEngine/web_engine_initialize.py b/sources/pyside2/tests/QtWebEngine/web_engine_initialize.py
index 1f0482f..f664d53 100644
--- a/sources/pyside2/tests/QtWebEngine/web_engine_initialize.py
+++ b/sources/pyside2/tests/QtWebEngine/web_engine_initialize.py
@@ -36,6 +36,9 @@ sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 from init_paths import init_test_paths
 init_test_paths(False)
 
+import ctypes
+ctypes.CDLL('libGLX_mesa.so.0', ctypes.RTLD_GLOBAL)
+
 from PySide2.QtCore import QTimer
 from PySide2.QtWebEngine import QtWebEngine
 from PySide2.QtWidgets import QApplication
